src/adv.py

```diff
@@ -49,3 +53,32 @@
 # Print an error message if the movement isn't allowed.
 #
 # If the user enters "q", quit the game.
+
+def different_way():
+    print('You cannot move that way, try a different way')
+    return
+
+def making_moves(player, direction):
+    attr = direction + '_to'
+
+    if hasattr(player.room, attr):
+        move_rooms = getattr(player.room, attr)
+        player.room = move_rooms
+        return
+    else:
+        different_way()
+
+playing = True
+while playing:
+    print('Location: ' + player.room.name)
+    print(player.room.description)
+    for x in player.room.items:
+        print(x.name)
+    direction = input('Which way? (n/e/s/w) or press q to quit\n=>')
+  
+    
+    if direction.lower().strip() == 'q':
+        print('goodbye')
+        playing = False
+    else: 
+        making_moves(player, direction)
```

Comment:
for adding functionality, this can be expanded into a long if/else if / else cascade (or even a switch statement).

Once you get to adding multi-line commands like "take item" you could try splitting the command into a list of words such as ["take","item"] and use the first item in the list to check what the command is, and the rest of the items in the list on how to handle that command inside its relative if/elseif block.